libs/challenge-registry/challenge-search/src/lib/challenge-search.component.ts

```diff
@@ -240,4 +254,31 @@ export class ChallengeSearchComponent implements OnInit, AfterContentInit {
       return true; // no filter applied
     }
   }
+
+  // mock up sorting challenges by certain property
+  private sortChallenges(
+    challenges: Challenge[],
+    sortBy: keyof Challenge | undefined
+  ): Challenge[] {
+    if (!sortBy) return challenges;
+
+    if (['startDate', 'endDate'].includes(sortBy as string)) {
+      // if it's starting soon, the status should be 'upcoming',
+      // otherwise, it's closing soon and status should be 'active',
+      const status = sortBy === 'startDate' ? 'upcoming' : 'active';
+      // sort challenges by startDate
+      // the sooner the challenge is going to start/end, the closer to the 1st card
+      // note: since it's a mock up func, the undefined of startDate/endDate is not considered here
+      return challenges
+        .filter((c) => c.status === status)
+        .sort(
+          (a, b) =>
+            +new Date(b.startDate as string) - +new Date(a.startDate as string)
```

Comment:
actually, if it's sorted by "Closing Soon", the "endDate" should be used to compare here...

Reply:
fixed